src/reducer.js

```diff
@@ -2,11 +2,16 @@
 import type { Action } from 'redux';
 import type { Location } from 'history';
 import { LOCATION_CHANGED } from './action-types';
+import isEqual from 'lodash.isequal';
 
 export default (state: ?Location | Object = {}, action: Action) => {
   if (action.type === LOCATION_CHANGED) {
     // No-op the initial route action
-    if (state && state.pathname === action.payload.pathname) {
+    if (
+      state &&
+      state.pathname === action.payload.pathname &&
+      isEqual(state.query, action.payload.query)
```

Comment:
I just realized...we have the raw query string in `state.search`, we can just compare those with `===`!